core/models/machine.py

```diff
@@ -301,15 +310,16 @@ def convert_glance_image(glance_image, provider_uuid, owner=None):
         identity = None
     app_kwargs.update({
         'created_by': user,
-        'created_by_identity': identity
+        'created_by_identity': identity,
+        'uuid': glance_image.get('application_uuid')
```

Comment:
Why are we setting the uuid from the glance metadata, what if doesn't exist in the metadata. Also what if an application exists with this uuid, it would result in an update to the existing app (so this outside image, would totally replace the existing one).

Reply:
For `glance images` that will be included in the Atmosphere image catalog,

If The `glance image` does not contain `application_uuid` in metadata,  it is the "Seed/First" version.
In this case, `create_application` will generate a UUID-hash with `glance_image.id`. Future lookups via monitoring can use `glance_image.id`  will always point to the created application.

Future updates to this image should come by the User issuing a `MachineRequest` to have a new image authored by admin identity.  After the MachineRequest is approved, the `processing` step of the imaging workflow is responsible for setting the metadata (`application_uuid, version_name, change_log, etc.`).

More updates on this process coming soon..